openslides_backend/action/actions/user/toggle_presence_by_number.py

```diff
@@ -0,0 +1,103 @@
+from ....permissions.management_levels import (
+    CommitteeManagementLevel,
+    OrganizationManagementLevel,
+)
+from ....permissions.permission_helper import (
+    has_committee_management_level,
+    has_organization_management_level,
+    has_perm,
+)
+from ....permissions.permissions import Permissions
+from ....shared.exceptions import ActionException, PermissionDenied
+from ....shared.filters import And, FilterOperator
+from ....shared.patterns import Collection, FullQualifiedId
+from ....shared.schema import required_id_schema
+from ...generics.update import UpdateAction
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionData
+
+
+@register_action("user.toggle_presence_by_number")
+class UserTogglePresenceByNumber(UpdateAction):
+    """
+    Action to toggle the presence by numbe
+    """
+
+    model = User()
+    schema = DefaultSchema(User()).get_default_schema(
+        additional_required_fields={
+            "number": {"type": "string"},
+            "meeting_id": required_id_schema,
+        }
+    )
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        """
+        update is_present_in_meeting_ids:
+        """
+        for instance in action_data:
+            meeting_id = instance.pop("meeting_id")
+            number = instance.pop("number")
+
+            instance["id"] = self.find_user_to_number(meeting_id, number)
+
+            user = self.datastore.get(
+                FullQualifiedId(self.model.collection, instance["id"]),
+                ["is_present_in_meeting_ids"],
+            )
+            is_present = user.get("is_present_in_meeting_ids", [])
+            if meeting_id not in is_present:
+                instance["is_present_in_meeting_ids"] = user.get(
+                    "is_present_in_meeting_ids", []
+                ) + [meeting_id]
+                yield instance
+            elif meeting_id in is_present:
+                is_present.remove(meeting_id)
+                instance["is_present_in_meeting_ids"] = is_present
+                yield instance
+
+    def find_user_to_number(self, meeting_id: int, number: str) -> int:
+        filter_ = FilterOperator(f"number_${meeting_id}", "=", number)
+        result = self.datastore.filter(Collection("user"), filter_, ["id"])
+        if len(result.keys()) == 1:
+            return list(result.keys())[0]
+        elif len(result.keys()) > 1:
+            raise ActionException("Found more than one user with the number.")
+
+        filter_ = And(
+            FilterOperator(f"number_${meeting_id}", "=", ""),
+            FilterOperator("default_number", "=", number),
+        )
+        result = self.datastore.filter(Collection("user"), filter_, ["id"])
+        if len(result.keys()) == 1:
+            return list(result.keys())[0]
+        raise ActionException("Could not find only one user with the number.")
```

Comment:
Differentiate here between the cases "No user found" and "Too many users found" for better error messages

Reply:
Diff the two cases.